tests/run-pass/malloc.rs

```diff
@@ -7,6 +8,16 @@ use core::{slice, ptr};
 extern crate libc;
 
 fn main() {
+    // Test that small allocations sometimes *are* not very aligned.
+    let saw_unaligned = (0..64).any(|_| unsafe {
+        let p = libc::malloc(3);
+        let addr = p as usize;
+        let unaligned = addr % 4 != 0; // test that this is not 4-aligned
+        libc::free(p); // FIXME have to free *after* test; should allow ptr-to-int of dangling ptr.
```

Comment:
Note: I should rebase before we land this; master will soon allow ptr-to-int for dangling pointers.